lib/FSi/Component/DataSource/Driver/Doctrine/ORM/DoctrineDriver.php

```diff
@@ -131,7 +131,16 @@ public function buildResult($fields, $first, $max)
             $this->currentQuery->setFirstResult($first);
         }
 
-        $result = new Paginator($this->currentQuery);
+        // Avoid DDC-2213 bug/mistake
```

Comment:
I think it would be much cleaner to put this logic into separate ``FSi\Component\DataSource\Driver\Doctrine\ORM\Paginator`` class which inherits from ``Doctrine\ORM\Tools\Pagination\Paginator``